src/config.rs

```diff
@@ -21,6 +22,19 @@ pub struct Config {
     #[validate(range(min = 0, max = 65535, message = "Invalid port", code = "bind_port"))]
     #[serde(default = "default_bind_port")]
     pub bind_port: u16,
+
+    #[serde(default = "default_certificate")]
+    pub certificate: String,
```

Comment:
I *think* you can use a `Path` here or similar and it will auto validate that it exists. But I could be thinking of a different library so ignore if so.

Reply:
So tried using path, but first rust said it didn't have fixed size so I needed to use &, then it said I something `expected named lifetime parameter` and I think my rust needs a level up first haha

Reply:
I've left it as string but I am using an Option for it in the config

Reply:
Oh, yeah Path is a trait I think (kinda like `interface` in Java), which is not sized. You would want `OsPath` or something, I haven't used that area of the stdlib much. String is definitely fine.